src/main/java/com/siit/hospital_manager/service/MedicationHistoryService.java

```diff
@@ -24,4 +29,15 @@ public List<MedicationHistoryDto> findAllByPatient(Integer id){
         return medicationHistoryRepository.findAllByPatient(zero).stream().map(MedicationHistoryDto::new).toList();
     }
 
+    public void addMedicationHistory (CreateMedicationHistoryDto dto, Integer patientId) {
+        MedicationHistory history = new MedicationHistory();
+        Patient patient = patientRepository.getById(patientId);
+        history.setMedication(dto.getMedication());
```

Comment:
all these fields should be checked for Not null & valid values before saving them to the database